File: clustermanager/src/nodes/connection/messages/request/request_serializer.go
package request

func SerializeRequest(request *Request) []byte {
	bytes := make([]byte, 0)

	bytes = append(bytes, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00)
	bytes = append(bytes, byte(len(request.authKey)<<2))
	bytes = append(bytes, []byte(request.authKey)...)
	bytes = append(bytes, request.operatorNumber<<2)
	bytes = append(bytes, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00)

	for _, arg := range request.args {
		bytes = append(bytes, byte(len(arg)))
		bytes = append(bytes, []byte(arg)...)
	}

	return bytes
}

File: core/src/messages/request/RequestSerializer.cpp
#include "messages/request/RequestSerializer.h"

std::vector<uint8_t> RequestSerializer::serialize(const Request& request) {
    std::vector<uint8_t> bytes{};

    Utils::appendToBuffer(request.requestNumber, bytes);

    bytes.push_back(request.authentication.authKey.size() << 2 | request.authentication.flag1 << 1 | request.authentication.flag2);
    Utils::appendToBuffer((uint8_t *) request.authentication.authKey.data(), request.authentication.authKey.size(), bytes);
    bytes.push_back(request.operation.operatorNumber << 2 | request.operation.flag1 << 1 | request.operation.flag2);
    Utils::appendToBuffer(request.operation.timestamp, bytes);
    if(request.authentication.flag1) //Includes selfNode
        Utils::appendToBuffer(request.operation.nodeId, bytes);

    for(auto i = request.operation.args->begin(); i < request.operation.args->end(); i++) {
        SimpleString<memDbDataLength_t> arg = * i;

        Utils::appendToBuffer(arg.size, bytes);
        Utils::appendToBuffer(arg.data(), arg.size, bytes);
    }

    return bytes;
}